libs/utils/executor.py

```diff
@@ -252,6 +252,14 @@ def _setup_cpufreq(self, tc):
 
         self.target.cpufreq.set_all_governors(cpufreq['governor'])
 
+        if 'freqs' in cpufreq:
+            if cpufreq['governor'] != 'userspace':
+                raise ValueError('Must use userspace governor to set CPU freqs')
+            logging.info(r'%14s - CPU frequencies: %s',
+                    'CPUFreq', str(cpufreq['freqs']))
+            for cpu, freq in cpufreq['freqs'].iteritems():
```

Comment:
Should not we add a check like:
```python
if len(cpufreq['freqs']) < len(self.target.list_online_cpus()):
    self._log.warning('Frequencies not defined for all CPUs')
```

Maybe the warning should be instead a raise ValueError?